source/extensions/http/header_validators/envoy_default/header_validator.cc

```diff
@@ -8,38 +14,344 @@ namespace EnvoyDefault {
 
 using ::envoy::extensions::http::header_validators::envoy_default::v3::HeaderValidatorConfig;
 using ::Envoy::Http::HeaderString;
+using ::Envoy::Http::Protocol;
+
+HeaderValidator::HeaderValidator(const HeaderValidatorConfig& config, Protocol protocol,
+                                 StreamInfo::StreamInfo& stream_info)
+    : config_(config), protocol_(protocol), stream_info_(stream_info),
+      header_values_(::Envoy::Http::Headers::get()) {}
+
+::Envoy::Http::HeaderValidator::HeaderEntryValidationResult
+HeaderValidator::validateMethodHeader(const HeaderString& value) {
+  // HTTP Method Registry, from iana.org:
+  // source: https://www.iana.org/assignments/http-methods/http-methods.xhtml
+  //
+  // From the RFC:
+  //
+  // tchar = "!" / "#" / "$" / "%" / "&" / "'" / "*" / "+" / "-" / "."
+  //       /  "^" / "_" / "`" / "|" / "~" / DIGIT / ALPHA
+  // token = 1*tchar
+  // method = token
+  //
+  static absl::node_hash_set<absl::string_view> kHttpMethodRegistry = {
```

Comment:
@yanavlasov do we need to have a hash set for the http_parser compatible methods or are we waiting to see if anyone complains about the base list being widened?

Reply:
We were not going to inherit existing code behavior that was not standard in the default header validator. We could have dedicated header validator that is http-parser compatible if there is a problem.